lib/validate.js

```diff
@@ -0,0 +1,108 @@
+'use babel'
+
+/* @flow */
+
+import invariant from 'assert'
+import type { Linter$UI, Linter$Linter, Linter$Message } from './types'
+
+export function ui(ui: Linter$UI): boolean {
+  let message
+
+  if (!ui || typeof ui !== 'object') {
+    message = 'UI Provider must be an object'
+  } else if (typeof ui.name !== 'string') {
+    message = 'UI.name must be a string'
+  } else if (typeof ui.activate !== 'function') {
+    message = 'UI.activate must be a function'
+  } else if (typeof ui.didCalculateMessages !== 'function') {
+    message = 'UI.didCalculateMessages must be a function'
+  } else if (typeof ui.didBeginLinting !== 'function') {
+    message = 'UI.didBeginLinting must be a function'
+  } else if (typeof ui.didFinishLinting !== 'function') {
+    message = 'UI.didFinishLinting must be a function'
+  } else if (typeof ui.dispose !== 'function') {
+    message = 'UI.dispose must be a function'
+  }
+
+  if (message) {
+    console.error('[Linter] Invalid UI provider received', message, ui)
+    throw new Error(message)
+  }
+
+  return true
+}
+
+export function linter(linter: Linter$Linter, indie: boolean = false): boolean {
+  let message
+
+  if (!linter || typeof linter !== 'object') {
+    message = 'Linter Provider must be an object'
+  } else if (linter.name && typeof linter.name !== 'string') {
```

Comment:
Yes it'll be undefined now, we previously executed validate all the time, so we expected stuff to be either defined or null. Now we only execute it in dev mode, which means we have adapted our codebase to work with undefines so null is no longer required